pkg/controllers/collaset/synccontrol/scale.go

```diff
@@ -89,3 +114,277 @@ func (s ActivePodsForDeletion) Less(i, j int) bool {
 	// TODO consider service available timestamps
 	return collasetutils.ComparePod(l.Pod, r.Pod)
 }
+
+// dealIncludeExcludePods returns pods which are allowed to exclude and include
+func (r *RealSyncControl) dealIncludeExcludePods(ctx context.Context, cls *appsv1alpha1.CollaSet, pods []*corev1.Pod) (sets.String, sets.String, error) {
+	if len(cls.Spec.ScaleStrategy.PodToExclude) == 0 && len(cls.Spec.ScaleStrategy.PodToInclude) == 0 {
+		return nil, nil, nil
+	}
+	ownedPods := sets.String{}
+	excludePodNames := sets.String{}
+	includePodNames := sets.String{}
+	for _, pod := range pods {
+		ownedPods.Insert(pod.Name)
+		if _, exist := pod.Labels[appsv1alpha1.PodExcludeIndicationLabelKey]; exist {
+			excludePodNames.Insert(pod.Name)
+		}
+	}
+
+	for _, podName := range cls.Spec.ScaleStrategy.PodToExclude {
+		if ownedPods.Has(podName) {
+			excludePodNames.Insert(podName)
+		}
+	}
+
+	for _, podName := range cls.Spec.ScaleStrategy.PodToInclude {
+		includePodNames.Insert(podName)
+	}
+
+	intersection := excludePodNames.Intersection(includePodNames)
+	if len(intersection) > 0 {
```

Comment:
I think we should reject this condition in validating webhook

Reply:
done